Test/Framework/Graphics/EffectTest.cs

```diff
@@ -100,5 +101,22 @@ public void EffectPassShouldOverrideTextureIfNotExplicitlySet()
             texture.Dispose();
             effect.Dispose();
         }
+
+        [Test]
+        public void EffectParameterShouldBeSetIfSetByNameAndGetByIndex()
+        {
+            // This relies on the parameters permanently being on the same index.
+            // Should be no problem except when adding parameters.
+            var effect = new BasicEffect(game.GraphicsDevice);
+            effect.TextureEnabled = true;
+            effect.Texture = null;
+            effect.Parameters["DiffuseColor"].SetValue(Color.HotPink.ToVector3());
+            effect.Parameters["FogColor"].SetValue(Color.Honeydew.ToVector3());
+
+            Assert.That(effect.Parameters[0].GetValueVector3().Equals(Color.HotPink.ToVector3()));
+            Assert.That(effect.Parameters[14].GetValueVector3().Equals(Color.Honeydew.ToVector3()));
+            
+            effect.Dispose();
```

Comment:
I would add a texture to this test as that is the other class of thing in parameters.

Reply:
Didn't need it for the test, so I deleted it.
Reverting.